### ISSUE TITLE
gradle builds: define Java language version 

### ISSUE THREAD

**Issue by MoritzKeppler** [microsoft/Data-Appliance-GX#83]
We should set a fixed java language version. Best would be to stick to latest LTS, i.e. java11.
Applies for gradle builds and docker file creation.
<sub> Moritz Keppler <moritz.keppler@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)</sub>

**Comment by DominikPinsel** [comment]
Another note from me: 
I just saw that the java version for the external:nifi:processors project needs to be java8. 

If I compile it with java11 and copy it to my local nifi docker image I get the following error:
`Failure to launch NiFi due to java.lang.UnsupportedClassVersionError: microsoft/dagx/transfer/nifi/processors/FetchS3Object has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0`

**Comment by jimmarino** [comment]
Yeah, unfortunately that seems to be one of the strange limitations with NiFi at the moment.

**Comment by beardyinc** [comment]
that's why in the CI build we use Java 8 as base image for the custom Nifi image. Check `.github/workflows/build-docker.yaml`, Line 36ff
